rr_viz/src/rr_viz/widgets/mission_dashboard/mapping/slam_supervisor.py

```diff
@@ -256,6 +261,45 @@ def delete_map_slot(self):
                     print("failed calling slam delete map service")
                     print(trig_resp.message)
 
+    def save_map_image(self):
+        map_name = self.mapName.text()
+
+        if map_name !='':
+            if map_name.find('.'):
+                map_name = string.replace(map_name, '.', '_')
+        else:
+            if not self.default_map_name:
+                map_name = randomTimeString()
+            else:
+                map_name = self.default_map_name
+                
+        if map_name =='':
+            if not self.default_map_name:
+                map_name = randomTimeString()
+            else:
+                map_name = self.default_map_name
+
+        if not os.path.exists(self.save_map_image_dir):
+            rospy.loginfo(self.save_map_image_dir + " doesn't exist, creating it now.")
+            os.mkdir(self.save_map_image_dir)
+
+        map_path = self.save_map_image_dir + '/' +  map_name
```

Comment:
Are we happy with overwriting the files that already exist? Just want to make sure that this is a decision we've taken.

Reply:
I presumed we were as that's what we do when we are saving a map. But maybe we shouldn't allow it without warning the user first? If that's the case I can add that functionality into the redesign of rr_viz I'm doing